From f9da7a51e4755c09433c02131135169b32c27f71 Mon Sep 17 00:00:00 2001
From: Daan De Meyer <daan.j.demeyer@gmail.com>
Date: Tue, 4 Feb 2025 20:46:11 +0100
Subject: [PATCH] test: Move external packages section down and reword a little

This is advanced level stuff that regular contributors don't care
about in the slightest, so move it further down. Also reword the
section a little while we're at it.

(cherry picked from commit f017a5521305987fdeadf85b975b8ed44a9950b6)
---
 test/README.testsuite | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/test/README.testsuite b/test/README.testsuite
index cec46d3e73..02f5404410 100644
--- a/test/README.testsuite
+++ b/test/README.testsuite
@@ -66,22 +66,6 @@ mkosi in the systemd reposistory, so any local modifications to the mkosi
 configuration (e.g. in `mkosi.local.conf`) are automatically picked up and used
 by the integration tests as well.
 
-## Running the integration tests without building systemd from source
-
-To make sure `mkosi` doesn't try to build systemd from source during the image build
-process, you can add the following to `mkosi.local.conf`:
-
-```
-[Build]
-Environment=NO_BUILD=1
-```
-
-You might also want to use the `PackageDirectories=` or `Repositories=` option to provide
-mkosi with a directory or repository containing the systemd packages that should be installed
-instead. If the repository containing the systemd packages is not a builtin repository known
-by mkosi, you can use the `SandboxTrees=` option to write an extra repository definition
-to /etc which is used when building the image instead.
-
 ## Iterating on an integration test
 
 To iterate on an integration test, let's first get a shell in the integration test environment by running
@@ -151,6 +135,22 @@ that make use of `run_testcases`.
 progress and only move the journal to its final location in the build directory
 (`$BUILD_DIR/test/journal`) when the test is finished.
 
+## Running the integration tests without building systemd from source
+
+If you want to run the integration tests against prebuilt systemd packages,
+first add the following to `mkosi.local.conf` to stop mkosi from building
+systemd packages from source:
+
+```conf
+[Build]
+Environment=NO_BUILD=1
+```
+
+You'll then probably want to use the `PackageDirectories=` or `SandboxTrees=`
+options to provide mkosi with a directory containing the systemd packages or a
+repository file that points to a repository with the systemd packages that
+should be installed.
+
 ### SELinux AVCs
 
 To have `TEST-06-SELINUX` check for SELinux denials, write the following to
